admin/base/utils.py

```diff
@@ -138,13 +138,20 @@ def change_embargo_date(registration, user, end_date):
 
     validate_embargo_date(registration, user, end_date)
 
-    registration._initiate_embargo(user, end_date,
-                                         for_existing_registration=True,
-                                         notify_initiator_on_complete=False)
-
-    if registration.is_public:
-        registration.is_public = False
-        registration.save()
+    if registration.embargo:
+        registration.embargo.end_date = end_date
+    else:
+        registration._initiate_embargo(
+            user,
+            end_date,
+            for_existing_registration=True,
+            notify_initiator_on_complete=False
+        )
+
+    registration.is_public = False
```

Comment:
Do you think there could be a case where an embargo end date is changed to the past, and therefore should be made public?

Reply:
Excellent point, currently I think that reg would be made public at midnight via the periodic task, but I should still put in some validation to prevent a admin from doing that accidentally.